### PR TITLE
feat: Allow to specify catalog source image for server:deploy && serv…

### PR DESCRIPTION
**Pull request by tolusha**
…er:update commands

### What does this PR do?
Allow to specify CatalogSource image with `--catalog-source-image` flag for `server:deploy` and `server:update` commands

### Screenshot/screencast of this PR

```
EXAMPLES
  # Update Eclipse Che:
  chectl server:update

  # Update Eclipse Che in 'eclipse-che' namespace:
  chectl server:update -n eclipse-che

  # Update Eclipse Che and update its configuration in the custom resource:
  chectl server:update --che-operator-cr-patch-yaml patch.yaml

  # Update Eclipse Che from the provided channel:
  chectl server:update --olm-channel next

  # Update Eclipse Che from the provided CatalogSource and channel:
  chectl server:update --olm-channel fast --catalog-source-name MyCatalogName --catalog-source-namespace MyCatalogNamespace

  # Create CatalogSource based on provided image and update Eclipse Che from it:
  chectl server:update --olm-channel latest --catalog-source-image MyCatalogImage

  # Create a CatalogSource defined in yaml file and update Eclipse Che from it:
  chectl server:update --olm-channel stable --catalog-source-yaml PATH_TO_CATALOG_SOURCE_YAML
```

### What issues does this PR fix or reference?

https://issues.redhat.com/browse/CRW-4075

### How to test this PR?

N/A

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

File: src/context.ts [hunk 1]
```diff
@@ -177,60 +179,53 @@ export namespace CheCtlContext {
...
-        ctx[EclipseCheContext.CATALOG_SOURCE_NAME] = EclipseChe.STABLE_CHANNEL_CATALOG_SOURCE
-      } else {
-        ctx[EclipseCheContext.CATALOG_SOURCE_NAME] = EclipseChe.NEXT_CHANNEL_CATALOG_SOURCE
-      }
+      ctx[EclipseCheContext.CATALOG_SOURCE_NAME] = ctx[EclipseCheContext.CHANNEL] === EclipseChe.STABLE_CHANNEL ? EclipseChe.STABLE_CHANNEL_CATALOG_SOURCE : EclipseChe.NEXT_CHANNEL_CATALOG_SOURCE
+    }
+
+    ctx[EclipseCheContext.CATALOG_SOURCE_IMAGE] = flags[CATALOG_SOURCE_IMAGE_FLAG]
+    if (ctx[EclipseCheContext.CATALOG_SOURCE_IMAGE]) {
+      ctx[EclipseCheContext.CATALOG_SOURCE_NAME] = EclipseChe.NEXT_CHANNEL_CATALOG_SOURCE
```
Comment (nickboldt) [inline src/context.ts:208]: I'm not sure this is correct... for dsc, I could be installing from quay.io/devspaces/iib:3.6.0.RC-04-27-v4.11-x86_64, which is technically the `latest` tag of the `fast` channel. Does `NEXT_CHANNEL_CATALOG_SOURCE` always equal `fast` ?

File: src/flags.ts [hunk 2]
```diff
@@ -228,67 +235,56 @@ export const STARTING_CSV = string({
...
 })
 
-export const PLATFORM_FLAG = 'platform'
-export const PLATFORM = string({
-  char: 'p',
-  description: 'Type of Kubernetes platform.',
-  options: ['minikube', 'k8s', 'openshift', 'microk8s', 'docker-desktop', 'crc'],
-  required: true,
+export const CATALOG_SOURCE_IMAGE =  string({
+  description: `OLM catalog source image to install ${EclipseChe.PRODUCT_NAME} operator.`,
```
Comment (nickboldt) [inline src/flags.ts:286]:
```suggestion
  description: `OLM catalog source image or IIB from which to install the ${EclipseChe.PRODUCT_NAME} operator.`,
```

File: src/flags.ts [hunk 3]
```diff
@@ -228,67 +235,56 @@ export const STARTING_CSV = string({
...
+  description: `Namespace for OLM catalog source to install ${EclipseChe.PRODUCT_NAME} operator.`,
   dependsOn: [CATALOG_SOURCE_NAME_FLAG],
-  exclusive: [CATALOG_SOURCE_YAML_FLAG],
+  exclusive: [CATALOG_SOURCE_YAML_FLAG, CATALOG_SOURCE_IMAGE_FLAG],
 })
 
 export const CATALOG_SOURCE_NAME =  string({
-  description: `OLM catalog source to install ${EclipseChe.PRODUCT_NAME} operator.
-                    This parameter is used only when the installer is the 'olm'.`,
+  description: `OLM catalog source name to install ${EclipseChe.PRODUCT_NAME} operator.`,
```
Comment (nickboldt) [inline src/flags.ts:280]: Does this have a default value, and is therefore not required? For example if I pass in `--catalog-source-image=quay.io/devspaces/iib:3.6.0.RC-04-27-v4.11-x86_64` could the catalog source name be generated as `iib-3-6-0-rc-04-27-v4-11-x86-64-fast` ... similar to https://github.com/redhat-developer/devspaces/blob/devspaces-3-rhel-8/product/installCatalogSourceFromIIB.sh#L262-L276 ? 

```suggestion
  description: `Name of the OLM catalog source or index bundle (IIB) from which to install ${EclipseChe.PRODUCT_NAME} operator.`,
```

File: src/context.ts (hunk 1 above)
Comment (tolusha) [reply src/context.ts:208]: It is `fast` for dsc [1]
https://github.com/redhat-developer/devspaces-chectl/blob/devspaces-3-rhel-8/build/scripts/sync.sh#L95-L106

File: src/flags.ts (hunk 3 above)
Comment (tolusha) [reply src/flags.ts]: It is `devspaces-fast`
https://github.com/redhat-developer/devspaces-chectl/blob/devspaces-3-rhel-8/build/scripts/sync.sh#L104

Comment (tolusha) [reply src/flags.ts]: But I agree it's better to generate a name.

File: src/context.ts (hunk 1 above)
Comment (nickboldt) [reply src/context.ts]: will I be able to use `server:update` to move from fast channel to stable channel, eg., for 

3.5.0.GA (stable) -> 3.6.0.RC (fast) -> 3.6.0.GA (stable)

**Pull request merged**